docs/reference/engine/schema/views.md

```diff
@@ -0,0 +1,113 @@
+---
+title: View entities
+---
+
+
+View entities in Contember enable developers to map SQL views to the Contember schema, providing a powerful tool for read-only queries. They follow specific rules and are useful in complex data representations.
+
+:::caution
+Using views in Contember provides a powerful way to define complex data relationships and aggregations. However, it requires an understanding of SQL, as you'll be writing direct SQL queries to shape the data.
+:::
+
+## Use Cases
+
+View entities are particularly useful in scenarios such as:
+
+- **Aggregated Reporting:** Creating summations or averages across multiple tables, e.g., sales reports.
+- **Data Transformation:** Displaying data in a transformed or customized manner, e.g., formatting values, processing JSONs
+- **Complex Join Operations:** Joining multiple tables in ways that would be cumbersome with traditional Contember entities, e.g., joining sales data with customer information.
+
+## Writing View Entities
+
+### 1. Defining the SQL Query
+
+The first step is to write the SQL query representing the view. Ensure it adheres to the following rules:
+
+- **Emit an `id` value**: Due to Contember's internal requirements, an `id` value must be included in the view. Typically, this is done using `gen_random_uuid() AS id` in the SQL.
+
+- **Field Naming Convention**: When defining fields in the SQL, use underscore_case (e.g., `total_count`). This will map to pascalCase in the Contember schema, ensuring proper correlation. For example, if you define a field as `total_count` in the SQL, you must name it the same way in the view schema.
```

Comment:
```suggestion
- **Field Naming Convention**: When defining fields in the SQL, use underscore_case (e.g., `total_count`). This will map to pascalCase in the Contember schema, ensuring proper correlation. For example, if you define a field as `total_count` in the SQL, you must name it `totalCount` in the view schema.
```